src/ui/components/DialogRelatedEntities/DialogRelatedEntities.tsx

```diff
@@ -0,0 +1,155 @@
+import {PlaceholderIllustration} from '../PlaceholderIllustration/PlaceholderIllustration';
+
+import {EntitiesList} from './EntitiesList/EntitiesList';
+
+import './DialogRelatedEntities.scss';
+
+const i18n = I18n.keyset('component.dialog-related-entities.view');
+
+const b = block('dialog-related-entities');
+
+type DialogRelatedEntitiesProps = EntryDialogProps & {
+    entry: GetEntryResponse;
+};
+
+enum Direction {
+    Parent = 'parent',
+    Child = 'child',
+}
+
+const CONCURRENT_ID = 'list-related-entities';
+
+export const DialogRelatedEntities = ({onClose, visible, entry}: DialogRelatedEntitiesProps) => {
+    const [currentDirection, setCurrentDirection] = React.useState<Direction>(
+        entry.scope === EntryScope.Dash ? Direction.Parent : Direction.Child,
+    );
+    const [isLoading, setIsLoading] = React.useState(true);
+    const [isError, setIsError] = React.useState(false);
+    const [relations, setRelations] = React.useState<Record<string, GetRelationsEntry[]> | null>(
+        null,
+    );
+    const [relationsCount, setRelationsCount] = React.useState<null | number>(null);
+
+    React.useEffect(() => {
+        setIsLoading(true);
+        setIsError(false);
+        getSdk().cancelRequest(CONCURRENT_ID);
+        getSdk()
+            .mix.getEntryRelations(
+                {
+                    entryId: entry.entryId,
+                    workbookId: entry.workbookId,
+                    direction: currentDirection,
+                },
+                {concurrentId: CONCURRENT_ID},
+            )
+            .then((response) => {
+                setRelationsCount(response.length);
+                setRelations(groupEntitiesByScope(response));
+                setIsLoading(false);
+            })
+            .catch((error) => {
+                if (error.isCancelled) {
+                    return;
+                }
+                setIsError(true);
+                setIsLoading(false);
+            });
+    }, [entry, currentDirection]);
+
+    const showDirectionControl =
+        entry.scope !== EntryScope.Dash && entry.scope !== EntryScope.Connection;
+
+    const handleDirectionParentdate = (value: Direction) => {
+        setCurrentDirection(value);
+    };
+
+    const handleClose = () => {
+        onClose({status: EntryDialogResolveStatus.Close});
+    };
+
+    const renderRelations = () => {
+        if (isLoading) {
+            return (
+                <div className={b('loader')}>
+                    <Loader />
+                </div>
+            );
+        }
+
+        if (
+            isEmpty(relations) &&
+            entry.scope === EntryScope.Widget &&
+            Object.values(EDITOR_TYPE).includes(entry.type) &&
+            currentDirection === Direction.Parent
+        ) {
+            return <Alert theme="warning" message={i18n('label_editor-hint')} />;
+        }
+
+        if (!relations || isEmpty(relations) || isError) {
```

Comment:
I think better to separate this if like this and move this logic right after isLoading check

```
if (isError) {
 return (
    <div className={b('error-state')}>
      <PlaceholderIllustration direction="column" name={'error'} title={i18n('label_request-error')} />
    </div>
 )
}
```

And `!relations  || isEmpty(relations)` when relations is [] | null is `isEmpty(relations)` (`isEmpty(null) -> true`)

In that case maybe:

```
if (isEmpty(relations)) {
   if (entry.scope === EntryScope.Widget &&
            Object.values(EDITOR_TYPE).includes(entry.type) &&
            currentDirection === Direction.Parent) { return <Alert/> }
  

  return (<div className={b('empty-state')}/>)
}
```